stl/src/filesystem.cpp

```diff
@@ -854,21 +825,45 @@ struct alignas(long long) _Aligned_file_attrs {
             _Flags, _Get_file_attributes_data)) { // caller wants something GetFileAttributesExW might provide
         if (_Symlink_attribute_hint == __std_fs_file_attr::_Invalid
             || !_Bitmask_includes(_Symlink_attribute_hint, __std_fs_file_attr::_Reparse_point)
-            || !_Follow_symlinks) { // we might not be a symlink or not following symlinks, so GetFileAttributesExW
+            || !_Follow_symlinks) { // we might not be a symlink or not following symlinks, so FindFirstFileW
                                     // would return the right answer
-            _Aligned_file_attrs _Aligned_attrs;
-            auto& _Data = _Aligned_attrs._Data;
-            if (!GetFileAttributesExW(_Path, GetFileExInfoStandard, &_Data)) {
-                return __std_win_error{GetLastError()};
+
+            // Check for file names that contain `?` or `*` (i.e., globbing characters).
+            // These are invalid file names, and will give us the wrong answer with `FindFirstFileW`.
+            {
+                const wchar_t* _Path_end           = _Path + _CSTD wcslen(_Path);
+                const wchar_t* _After_drive_prefix = _STD _Find_root_name_end(_Path, _Path_end);
+                // `?` is allowed in the drive prefix, but `*` is not.
+                if (_STD find(_Path, _After_drive_prefix, '*') != _After_drive_prefix) {
+                    return __std_win_error{ERROR_INVALID_NAME};
+                }
+
+                constexpr static auto _Is_globbing_character = [](wchar_t _Ch) { return _Ch == '*' || _Ch == '?'; };

```

Comment:
First, we conventionally say `static constexpr`.

Second, the `static` is unnecessary and can be omitted. (It's not wrong though, and you could keep it if you really want.)

`static` makes a big difference for arrays at local scope - a non-`static` array, even if it's `constexpr`, will generate code to push each value onto the stack. (This is because distinct objects need distinct addresses; the compiler can skip this if it can see that nobody observes the address, but arrays are inherently almost always address-taken.) `static constexpr` arrays will be put into read-only memory so the only codegen is pushing a pointer to that memory.

But for scalars (and lambdas), there is no codegen difference when adding `static` here.